src/frankenphp-symfony/src/Runner.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Runtime\FrankenPhpSymfony;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\HttpKernel\TerminableInterface;
+use Symfony\Component\Runtime\RunnerInterface;
+
+/**
+ * A runner for FrankenPHP.
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ */
+class Runner implements RunnerInterface
+{
+    public function __construct(private readonly HttpKernelInterface $kernel)
+    {
+    }
+
+    public function run(): int
+    {
+        $server = array_filter($_SERVER, static fn (string $key) => !str_starts_with($key, 'HTTP_'), ARRAY_FILTER_USE_KEY);
+        do {
+            $ret = frankenphp_handle_request(function () use ($server, &$sfRequest, &$sfResponse) {
+                // Merge the environment variables coming from DotEnv with the one tight to the current request
+                $_SERVER += $server;
+
+                $sfRequest = Request::createFromGlobals();
+                $sfResponse = $this->kernel->handle($sfRequest);
+
+                $sfResponse->send();
+            });
+
+            if ($this->kernel instanceof TerminableInterface && $sfRequest && $sfResponse) {
+                $this->kernel->terminate($sfRequest, $sfResponse);
+            }
```

Comment:
Should this part not be also inside the `frankenphp_handle_request`. Or does it have no difference here? If its inside or outside the callable. I'm not sure if the terminate is this way called for every request as it should, else I know some projects which would crash into unexpected state.

Reply:
Frankenphp only cares about putting out the response. A "cleanup" or background process could be running outside IMHO.

Reply:
> I'm not sure if the terminate is this way called for every request as it should

It does.